conf/app.routes

```diff
@@ -1,3 +1,13 @@
 # microservice specific routes
 
-POST        /id         uk.gov.hmrc.apisubscriptionfields.controller.ApiSubscriptionFieldsController.id
+# TODO add endpoints for field definitions CRUD operations
+
+GET         /id/:appId/:apiContext/:apiVersion                                 @uk.gov.hmrc.apisubscriptionfields.controller.ApiSubscriptionFieldsController.id(appId: java.util.UUID, apiContext: String, apiVersion: String)
+
+GET         /application/:appId/context/:apiContext/version/:apiVersion        @uk.gov.hmrc.apisubscriptionfields.controller.ApiSubscriptionFieldsController.getSubscriptionFields(appId: java.util.UUID, apiContext: String, apiVersion: String)
+
+GET         /fieldsId/:id                                                      @uk.gov.hmrc.apisubscriptionfields.controller.ApiSubscriptionFieldsController.getSubscriptionFieldsByFieldsId(id: java.util.UUID)
+
+PUT         /application/:appId/context/:apiContext/version/:apiVersion        @uk.gov.hmrc.apisubscriptionfields.controller.ApiSubscriptionFieldsController.upsertSubscriptionFields(appId: java.util.UUID, apiContext: String, apiVersion: String)
+
+DELETE      /application/:appId/context/:apiContext/version/:apiVersion        @uk.gov.hmrc.apisubscriptionfields.controller.ApiSubscriptionFieldsController.deleteSubscriptionFields(appId: java.util.UUID, apiContext: String, apiVersion: String)
```

Comment:
Do we need to call this `DELETE` endpoint from other microservices? 
Or is it just an admin task we want to add.
We have two indexes in the mongo collection:
- `id` (applicationId + apiContext + apiVersion)
- `fieldsId`
It is simpler to delete by `fieldsId`. In fact the endpoint would be:
```
DELETE      /fieldsId/:fieldsId 
```

Reply:
Ticket mentions this endpoint as it is currently implemented. Happy to change if ticket is amended.

Reply:
It was a suggestion, to avoid confusion.
The ticket description might need adjustments as well

Reply:
Sure, can we wait until Tim updates the ticket?

Reply:
Was intended to be used if an API was retired.  Not needed yet because there will not be many APIs using the custom fields but it would be nice to have for when the retired workflow is updated.  If we use fieldsId as the key, the publisher would have to do two calls because it does not have the fieldsId, only the application id, api name and version.

Reply:
I reviewed this
Both `id` and `fieldsId` are indexes, and they are a unique field in the collection.
Let's follow @crystalmark idea